bin/improver-spot-extract-new

```diff
@@ -0,0 +1,130 @@
+                             "data to be extracted.")
+    parser.add_argument("output_filepath", metavar="OUTPUT_FILEPATH",
+                        help="The output path for the resulting NetCDF")
+
+    method_group = parser.add_argument_group("Neighbour finding method")
+    method_group.add_argument(
+        "--land_constraint", default=False, action='store_true',
+        help="If set the neighbour cube will be interogated for grid point"
+        " neighbours that were identified using a land constraint. This means"
+        " that the grid points should be land points except for sites where"
+        " none was found within the search radius when the neighbour cube was"
+        " created. May be used with minimum_dz.")
+    method_group.add_argument(
+        "--minimum_dz", default=False, action='store_true',
+        help="If set the neighbour cube will be interogated for grid point"
+        " neighbours that were identified using a minimum height difference"
+        " constraint. These are grid points that were found to be the closest"
+        " in altitude to the spot site within the search radius defined when"
+        " the neighbour cube was created. May be used with land_constraint.")
+
+    lapse_group = parser.add_argument_group(
+        "Temperature lapse rate adjustment")
+    lapse_group.add_argument(
+        "--temperature_lapse_rate_filepath",
+        help="Filepath to a NetCDF file containing temperature lapse rates. "
+        "If this cube is provided, and a screen temperature cube is being "
+        "processed, the lapse rates will be used to adjust the temperatures "
+        "to better represent each spot site's altitude.")
+
+    meta_group = parser.add_argument_group("Metadata")
+    meta_group.add_argument(
+        "--grid_metadata_identifier", default="mosg",
+        help="A string to identify attributes from the input netCDF files that"
+        " should be compared to ensure that the data is compatible. Spot"
+        " data works using grid indices, so it is important that the grids"
+        " are matching or the data extracted may not match the location of the"
+        " spot data sites.")
+
+    args = parser.parse_args()
+    neighbour_cube = load_cube(args.neighbour_filepath)
+    diagnostic_cube = load_cube(args.diagnostic_filepath)
+
+    neighbour_selection_method = NeighbourSelection(
+        land_constraint=args.land_constraint,
+        minimum_dz=args.minimum_dz).neighbour_finding_method_name()
+
+    plugin = SpotExtraction(
+        neighbour_selection_method=neighbour_selection_method,
+        grid_metadata_identifier=args.grid_metadata_identifier)
+    result = plugin.process(neighbour_cube, diagnostic_cube)
+
+    # Check whether the diagnostic is screen temperature and a lapse rate cube
+    # has been provided.
+    screen_temperature = (
+        diagnostic_cube.name() == "air_temperature" and
+        np.isclose(diagnostic_cube.coord("height").points, 1.5) and
```

Comment:
Can you explain this height criterion please?  Why do we need it and is there a way to avoid hard-coding the value in this way?

Reply:
Actually, I may just test that it matches the height in the lapse rate cube and is single valued. Then if someone creates a lapse rate cube at 2m, because that's the height they care about, this will just check they are using temperature data that matches.